File: mpp/Lab8/csharpPb9/networking/DTO/UserDTO.cs
using System;
using System.Collections.Generic;
using System.Data.Common;
using System.Linq;
using System.Text;
using System.Text.Json.Serialization;
using System.Threading.Tasks;

namespace networking.DTO
{
    [Serializable]
    internal class UserDTO
    {
        [JsonPropertyName("id")]
        public long id { get; set; }

        [JsonPropertyName("nume")]
        public string nume { get; set; }

        [JsonPropertyName("username")]
        public string username { get; set; }

        [JsonPropertyName("password")]
        public string password { get; set; }

        [JsonPropertyName("id_proba")]
        public string id_proba { get; set; }
        public UserDTO() { }

        public UserDTO(long id, string nume, string username, string password, string id_proba)
        {
            this.id = id;
            this.nume = nume;
            this.username = username;
            this.password = password;
            this.id_proba = id_proba;
        }

        public UserDTO(string nume, string username, string password, string id_proba)
            : this(0, nume, username, password, id_proba)
        {}

        public override string ToString()
        {
            return $"UserDTO [id={id}, nume={nume}, username={username}, password={password}, id_proba={id_proba}]";
        }
    }
}

File: mpp/Lab10/javaPb9/networking/src/main/java/ro/mpp2024/DTO/UserDTO.java
package ro.mpp2024.DTO;

import java.io.Serializable;

public class UserDTO implements Serializable {
    private Long id;
    private String nume;
    private String username;
    private String password;
    private String id_proba;

    public UserDTO() {
    }

    public UserDTO(Long id,String nume, String username, String password, String probaId) {
        this.id = id;
        this.nume = nume;
        this.username = username;
        this.password = password;
        this.id_proba = probaId;
    }

    public UserDTO(String nume, String username, String password, String probaId) {
        id = -1L;
        this.nume = nume;
        this.username = username;
        this.password = password;
        this.id_proba = probaId;
    }

    public Long getId() {
        return id;
    }

    public String getNume() {
        return nume;
    }

    public String getUsername() {
        return username;
    }

    public String getPassword() {
        return password;
    }

    public String getProbaId() {
        return id_proba;
    }

    @Override
    public String toString() {
        return "UserDTO{" +
                "nume='" + nume + '\'' +
                ", username='" + username + '\'' +
                ", password='" + password + '\'' +
                ", probaId='" + id_proba + '\'' +
                '}';
    }
}
